### PR TITLE
add e2e test for release pipeline runner

### PR DESCRIPTION
**Pull request by riarenas**
fixes https://github.com/dotnet/arcade/issues/2319

Runs a very simple release pipeline https://dnceng.visualstudio.com/internal/_release?view=all&_a=releases&definitionId=45 and waits for PR creation.

### CODE REVIEW

**Comment by JohnTortugo** [comment]
Hey Ricardo, is there a place where I can see this running before the merge?

**Comment by riarenas** [comment]
None of the tests under the scenario folder run automatically yet. I'm also working on hooking these up to the release. (they will never run before a merge automatically because we'd need to deploy the whole service fabric and programmatically get an api token for that service fabric instance)

File: src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1 [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            if ($pullRequest.title -ne $expectedPRTitle) {
+                throw "Expected PR title to be $expectedPRTitle, was $($pullRequest.title)"
+            }
+            
+            # Check out the merge commit sha, then use darc to get and verify the
+            # dependencies
+            Git-Command $targetRepoName fetch
+            Git-Command $targetRepoName checkout $pullRequestBaseBranch
+
+            try {
```
Comment (mmitche) [inline src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1]: I don't see where you're checking for the AzDO release here, did I miss it? (looking for Get-AzDO-Release)

Comment (riarenas) [reply src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1]: I actually meant to delete that function, as I haven't been able to get the authentication for this API quite right.

I thought for a minute that it'd be enough to check whether the PR gets opened, but on further thought that won't be enough. I'm working on getting the auth right now, and will add a proper check to see if a release for the bar build ID was triggered.

File: src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1 [hunk 2]
```diff
@@ -169,7 +191,7 @@ try {
     if ($buildInfo.channels.length -ne 1) {
         throw "Expected to see build in 1 channel, got $($buildInfo.channels.length)"
     }
-    $success = ($success) -and ($buildInfo.channels[0].name -eq $testChannelName)
+    $success =  ($buildInfo.channels[0].name -eq $testChannelName)
```
Comment (mjanecke) [inline src/Maestro/tests/Scenarios/githubflow-release-pipeline-nonbatched.ps1]: nit: extra space after =

**Comment by riarenas** [comment]
Added the functionality to check that the release was actually triggered. It will only work while running in Azure DevOps since that API requires OAuth2 credentials flow. Azure devops already provides a $system.accessToken variable that solves this.

I hacked up a release definition that runs this test: https://dev.azure.com/dnceng/internal/_releaseProgress?_a=release-environment-logs&releaseId=4722&environmentId=10877

The overall test failed because of https://github.com/dotnet/arcade/issues/2529, but everything up until the checks to find the release succeeded.

**Comment by riarenas** [comment]
Test release that ran this successfully: https://dev.azure.com/dnceng/internal/_releaseProgress?_a=release-environment-logs&releaseId=4766&environmentId=10987

**Pull request merged**